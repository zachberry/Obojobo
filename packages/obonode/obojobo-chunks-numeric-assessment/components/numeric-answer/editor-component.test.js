import React from 'react'
import renderer from 'react-test-renderer'
import { mount } from 'enzyme'

import NumericAnswer from './editor-component'

describe('NumericAnswer Editor Node', () => {
	test('NumericAnswer builds the expected component with score of 100', () => {
		const component = renderer.create(
			<NumericAnswer
				node={{
					data: {
						get: () => {
							return {
								score: 100
							}
						}
					}
				}}
			/>
		)
		const tree = component.toJSON()

		expect(tree).toMatchSnapshot()
	})

	test('NumericAnswer builds the expected component with score of 0', () => {
		const component = renderer.create(
			<NumericAnswer
				node={{
					data: {
						get: () => {
							return {
								score: 0
							}
						}
					}
				}}
			/>
		)
		const tree = component.toJSON()

		expect(tree).toMatchSnapshot()
	})

	test('NumericAnswer onHandleScoreChange() with score of `100`', () => {
		const editor = {
			setNodeByKey: jest.fn()
		}

		const component = mount(
			<NumericAnswer
				node={{
					key: 'mockKey',
					nodes: [],
					data: {
						get: () => {
							return {
								score: 100
							}
						}
					}
				}}
				editor={editor}
			/>
		)

		component
			.find('button')
			.at(0)
			.simulate('click')

		const tree = component.html()

		expect(editor.setNodeByKey).toHaveBeenCalled()
		expect(tree).toMatchSnapshot()
	})

	test('NumericAnswer onHandleScoreChange() with score `0`', () => {
		const editor = {
			setNodeByKey: jest.fn()
		}

		const component = mount(
			<NumericAnswer
				node={{
					key: 'mockKey',
					nodes: [],
					data: {
						get: () => {
							return {
								score: 0
							}
						}
					}
				}}
				editor={editor}
			/>
		)

		component
			.find('button')
			.at(0)
			.simulate('click')

		const tree = component.html()

		expect(editor.setNodeByKey).toHaveBeenCalled()
		expect(tree).toMatchSnapshot()
	})

	test('NumericAnswer onClickDropdown() change requirement from `Exact answer` to `Margin of error`', () => {
		const editor = {
			setNodeByKey: jest.fn()
		}

		const component = mount(
			<NumericAnswer
				node={{
					key: 'mockKey',
					nodes: [],
					data: {
						get: () => {
							return {
								requirement: 'exact',
								score: 0
							}
						}
					}
				}}
				editor={editor}
			/>
		)

		component
			.find('.select-item')
			.at(0)
			.simulate('change', { target: { name: 'requirement', value: 'Margin of error' } })

		const tree = component.html()

		expect(editor.setNodeByKey).toHaveBeenCalled()
		expect(tree).toMatchSnapshot()
	})

	test('NumericAnswer onClickDropdown() change requirement from `Exact answer` to `Precise response`', () => {
		const editor = {
			setNodeByKey: jest.fn()
		}

		const component = mount(
			<NumericAnswer
				node={{
					key: 'mockKey',
					nodes: [],
					data: {
						get: () => {
							return {
								requirement: 'exact',
								score: 0
							}
						}
					}
				}}
				editor={editor}
			/>
		)

		component
			.find('.select-item')
			.at(0)
			.simulate('change', { target: { name: 'requirement', value: 'Precise response' } })

		const tree = component.html()

		expect(editor.setNodeByKey).toHaveBeenCalled()
		expect(tree).toMatchSnapshot()
	})

	test('NumericAnswer onClickDropdown() change `type` for `Margin of error`', () => {
		const editor = {
			setNodeByKey: jest.fn()
		}

		const component = mount(
			<NumericAnswer
				node={{
					key: 'mockKey',
					nodes: [],
					data: {
						get: () => {
							return {
								requirement: 'margin',
								score: 0
							}
						}
					}
				}}
				editor={editor}
			/>
		)

		component
			.find('.select-item')
			.at(0)
			.simulate('change', { target: { name: 'type', value: 'percent' } })

		const tree = component.html()

		expect(editor.setNodeByKey).toHaveBeenCalled()
		expect(tree).toMatchSnapshot()
	})

	test('NumericAnswer onHandleInputChange() works as expected', () => {
		const editor = {
			setNodeByKey: jest.fn()
		}

		const component = mount(
			<NumericAnswer
				node={{
					key: 'mockKey',
					nodes: [],
					data: {
						get: () => {
							return {
								requirement: 'exact',
								score: 0
							}
						}
					}
				}}
				editor={editor}
			/>
		)

		component
			.find('.input-item')
			.at(0)
			.simulate('change', { target: { name: 'value', value: '100' } })

		const tree = component.html()

		expect(editor.setNodeByKey).toHaveBeenCalled()
		expect(tree).toMatchSnapshot()
	})
})
