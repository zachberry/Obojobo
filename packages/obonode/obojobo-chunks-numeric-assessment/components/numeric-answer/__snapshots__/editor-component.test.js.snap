// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NumericAnswer Editor Node NumericAnswer builds the expected component with score of 0 1`] = `
<div
  className="numeric-input-container pad"
>
  <button
    className="correct-button is-not-correct"
    onClick={[Function]}
    tabIndex="0"
  >
    ✖
  </button>
  <table
    contentEditable={false}
  >
    <thead>
      <tr>
        <th>
          Requirement
        </th>
        <th>
          Answer
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td>
          <select
            className="select-item"
            name="requirement"
            onChange={[Function]}
            onClick={[Function]}
          >
            <option>
              Exact answer
            </option>
            <option>
              Margin of error
            </option>
            <option>
              Within a range
            </option>
            <option>
              Precise response
            </option>
          </select>
        </td>
        <td>
          <input
            className="input-item"
            contentEditable={false}
            name="answer"
            onChange={[Function]}
            onClick={[Function]}
            value=""
          />
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;

exports[`NumericAnswer Editor Node NumericAnswer builds the expected component with score of 100 1`] = `
<div
  className="numeric-input-container pad"
>
  <button
    className="correct-button is-correct"
    onClick={[Function]}
    tabIndex="0"
  >
    ✔
  </button>
  <table
    contentEditable={false}
  >
    <thead>
      <tr>
        <th>
          Requirement
        </th>
        <th>
          Answer
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td>
          <select
            className="select-item"
            name="requirement"
            onChange={[Function]}
            onClick={[Function]}
          >
            <option>
              Exact answer
            </option>
            <option>
              Margin of error
            </option>
            <option>
              Within a range
            </option>
            <option>
              Precise response
            </option>
          </select>
        </td>
        <td>
          <input
            className="input-item"
            contentEditable={false}
            name="answer"
            onChange={[Function]}
            onClick={[Function]}
            value=""
          />
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;

exports[`NumericAnswer Editor Node NumericAnswer onClickDropdown() change \`type\` for \`Margin of error\` 1`] = `"<div class=\\"numeric-input-container pad\\"><button class=\\"correct-button is-not-correct\\" tabindex=\\"0\\">✖</button><table contenteditable=\\"false\\"><thead><tr><th>Requirement</th><th>Type</th><th>Answer</th><th>Margin</th></tr></thead><tbody><tr><td><select class=\\"select-item\\" name=\\"requirement\\"><option>Exact answer</option><option>Margin of error</option><option>Within a range</option><option>Precise response</option></select></td><td><select class=\\"select-item\\" name=\\"type\\"><option>Percent</option><option>Absolute</option></select></td><td><input class=\\"input-item\\" name=\\"answer\\" value=\\"\\"></td><td><input class=\\"input-item\\" name=\\"margin\\" value=\\"\\"></td></tr></tbody></table></div>"`;

exports[`NumericAnswer Editor Node NumericAnswer onClickDropdown() change requirement from \`Exact answer\` to \`Margin of error\` 1`] = `"<div class=\\"numeric-input-container pad\\"><button class=\\"correct-button is-not-correct\\" tabindex=\\"0\\">✖</button><table contenteditable=\\"false\\"><thead><tr><th>Requirement</th><th>Answer</th></tr></thead><tbody><tr><td><select class=\\"select-item\\" name=\\"requirement\\"><option>Exact answer</option><option>Margin of error</option><option>Within a range</option><option>Precise response</option></select></td><td><input class=\\"input-item\\" name=\\"answer\\" contenteditable=\\"false\\" value=\\"\\"></td></tr></tbody></table></div>"`;

exports[`NumericAnswer Editor Node NumericAnswer onClickDropdown() change requirement from \`Exact answer\` to \`Precise response\` 1`] = `"<div class=\\"numeric-input-container pad\\"><button class=\\"correct-button is-not-correct\\" tabindex=\\"0\\">✖</button><table contenteditable=\\"false\\"><thead><tr><th>Requirement</th><th>Answer</th></tr></thead><tbody><tr><td><select class=\\"select-item\\" name=\\"requirement\\"><option>Exact answer</option><option>Margin of error</option><option>Within a range</option><option>Precise response</option></select></td><td><input class=\\"input-item\\" name=\\"answer\\" contenteditable=\\"false\\" value=\\"\\"></td></tr></tbody></table></div>"`;

exports[`NumericAnswer Editor Node NumericAnswer onHandleInputChange() works as expected 1`] = `"<div class=\\"numeric-input-container pad\\"><button class=\\"correct-button is-not-correct\\" tabindex=\\"0\\">✖</button><table contenteditable=\\"false\\"><thead><tr><th>Requirement</th><th>Answer</th></tr></thead><tbody><tr><td><select class=\\"select-item\\" name=\\"requirement\\"><option>Exact answer</option><option>Margin of error</option><option>Within a range</option><option>Precise response</option></select></td><td><input class=\\"input-item\\" name=\\"answer\\" contenteditable=\\"false\\" value=\\"\\"></td></tr></tbody></table></div>"`;

exports[`NumericAnswer Editor Node NumericAnswer onHandleScoreChange() with score \`0\` 1`] = `"<div class=\\"numeric-input-container pad\\"><button class=\\"correct-button is-not-correct\\" tabindex=\\"0\\">✖</button><table contenteditable=\\"false\\"><thead><tr><th>Requirement</th><th>Answer</th></tr></thead><tbody><tr><td><select class=\\"select-item\\" name=\\"requirement\\"><option>Exact answer</option><option>Margin of error</option><option>Within a range</option><option>Precise response</option></select></td><td><input class=\\"input-item\\" name=\\"answer\\" contenteditable=\\"false\\" value=\\"\\"></td></tr></tbody></table></div>"`;

exports[`NumericAnswer Editor Node NumericAnswer onHandleScoreChange() with score of \`100\` 1`] = `"<div class=\\"numeric-input-container pad\\"><button class=\\"correct-button is-correct\\" tabindex=\\"0\\">✔</button><table contenteditable=\\"false\\"><thead><tr><th>Requirement</th><th>Answer</th></tr></thead><tbody><tr><td><select class=\\"select-item\\" name=\\"requirement\\"><option>Exact answer</option><option>Margin of error</option><option>Within a range</option><option>Precise response</option></select></td><td><input class=\\"input-item\\" name=\\"answer\\" contenteditable=\\"false\\" value=\\"\\"></td></tr></tbody></table></div>"`;
